go/vt/tabletserver/mock/controller.go

```diff
@@ -0,0 +1,116 @@
+// Copyright 2015, Google Inc. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// Package mock provides mock interfaces for tabletserver.
+package mock
+
+import (
+	"github.com/youtube/vitess/go/vt/dbconfigs"
+	"github.com/youtube/vitess/go/vt/mysqlctl"
+	pb "github.com/youtube/vitess/go/vt/proto/query"
+	"github.com/youtube/vitess/go/vt/proto/topodata"
+	"github.com/youtube/vitess/go/vt/tabletserver"
+	"github.com/youtube/vitess/go/vt/tabletserver/queryservice"
+)
+
+// Controller is a fake version of tabletserver.Controller
+type Controller struct {
+	// QueryServiceEnabled is a state variable
+	QueryServiceEnabled bool
+
+	// InitDBConfigError is the return value for InitDBConfig
+	InitDBConfigError error
+
+	// SetServingTypeError is the return value for SetServingType
+	SetServingTypeError error
+
+	// StartServiceError is the return value for StartService
+	StartServiceError error
+
+	// IsHealthy is the return value for IsHealthy
+	IsHealthyError error
+
+	// ReloadSchemaCount counts how many times ReloadSchema was called
+	ReloadSchemaCount int
+}
+
+// NewController returns an implementation of tabletserver.Controller
+// that is entirely fake
```

Comment:
So is this a fake or a mock? I would say fake.

Then, fakets as the package name?